src/routes/api/users/getById.js

```diff
@@ -1,16 +1,57 @@
 const logger = require("../../../logger");
-const { createSuccessResponse } = require("../../../response");
+const hash = require("../../../hash");
+const validator = require("validator");
+const {
+  createSuccessResponse,
+  createErrorResponse,
+} = require("../../../response");
+const { PrismaClient } = require("@prisma/client");
+const prisma = new PrismaClient();
 
 /**
  * Get a user by their id
  */
-
-// eslint-disable-next-line no-unused-vars
-module.exports = async (req, res, next) => {
+module.exports = async (req, res) => {
   logger.debug(
     { user: req.user, id: req.params.id },
     `received request: GET /v1/users/:userId`,
   );
 
-  return res.status(418).json(createSuccessResponse());
+  const email = req.params.userId;
+  if (!validator.isEmail(email)) {
+    return res
+      .status(400)
+      .json(createErrorResponse(400, "Invalid email format"));
+  }
+
+  const userId = hash(email).toString();
+
+  try {
+    const user = await prisma.user.findUnique({
+      where: { hashedEmail: userId }, //searching prisma for the userId (hashed email)
+      select: {
+        id: true,
+        hashedEmail: true,
+        joinDate: true,
+        displayName: true,
+        profileBio: true,
+        profilePictureUrl: true,
+        profilePictureUrlExpiry: true,
+        assets: true,
```

Comment:
question: Should we use `include` for the relation fields? https://www.prisma.io/docs/orm/prisma-client/queries/select-fields#overview